bftengine/src/bftengine/ClientsManager.cpp

```diff
@@ -24,6 +24,158 @@
 using namespace std::chrono;
 using namespace concord::serialize;
 namespace bftEngine::impl {
+
+/*************************** Class ClientsManager::RequestsInfo ***************************/
+
+void ClientsManager::RequestsInfo::emplaceSafe(NodeIdType clientId, ReqId reqSeqNum, const std::string& cid) {
+  const lock_guard<mutex> lock(requestsMapMutex_);
+  if (requestsMap_.find(reqSeqNum) != requestsMap_.end()) {
+    LOG_WARN(CL_MNGR, "The request already exists - skip adding" << KVLOG(clientId, reqSeqNum));
+    return;
+  }
+  requestsMap_.emplace(reqSeqNum, RequestInfo{getMonotonicTime(), cid});
+  LOG_DEBUG(CL_MNGR, "Added request" << KVLOG(clientId, reqSeqNum, requestsMap_.size()));
+}
+
+bool ClientsManager::RequestsInfo::findSafe(ReqId reqSeqNum) {
+  const lock_guard<mutex> lock(requestsMapMutex_);
+  return (requestsMap_.find(reqSeqNum) != requestsMap_.end());
+}
+
+bool ClientsManager::RequestsInfo::removeRequestsOutOfBatchBoundsSafe(NodeIdType clientId, ReqId reqSequenceNum) {
```

Comment:
Thanks Yulia for creating safe methods for the thread-unsafe methods. Any reasons for not calling the unsafe methods by putting the lock in safe method. That would avoid code duplication?

Reply:
Oh, I notice you have removed unsafe methods. This is fine. Thanks.